### PR TITLE
2.4.14 version update, jsvm, callStatic Method, etc

### PR DESCRIPTION
**Pull request by wanghui187**
1. Performance related:
a. Adapting to HarmonyOS's built-in JS engine (JSVM) can support JIT and improve performance
b. Adding an eggsurface recycling mechanism can effectively reduce the probability of games being killed in the background
c. Modify rendering load configuration, reduce GPU bandwidth, lower power consumption, and improve performance
2. Function related
a. By default, when compiling under C++debug, add debugging information to facilitate C++breakpoints
b. Modify the engine exit method to make the exit more elegant
c. Jsb.reflection.callStatisticMethod implementation in HarmonyOS

### CODE REVIEW

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +68,7 @@ find_library( # Sets the name of the path variable.
               rawfile.z )
 message(STATUS, "rawfile=${rawfile-lib}")
 target_link_libraries(${CC_LIB_NAME} PUBLIC
-${EGL-lib} ${GLESv3-lib} ${hilog-lib} ${libace-lib} ${libnapi-lib} ${libuv-lib} ${rawfile-lib} libnative_drawing.so libc++.a libOpenSLES.so libohaudio.so libavplayer.so
+${EGL-lib} ${GLESv3-lib} ${hilog-lib} ${libace-lib} ${libnapi-lib} ${libuv-lib} ${rawfile-lib} libnative_drawing.so libc++.a libOpenSLES.so libohaudio.so libavplayer.so libjsvm.so libnative_window.so libnative_buffer.so  libbundle_ndk.z.so
```
Comment (qiuguohua) [inline templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt:71]: Should libjsvm.so not need to be linked when not using jsvm?

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt [hunk 2]
```diff
@@ -7,14 +7,19 @@ set(CC_COCOS_SOURCES)
 set(CC_COMMON_SOURCES)
 set(CC_ALL_SOURCES)
 
-set(CC_USE_SE_V8 ON)
+set(CC_USE_SE_V8 OFF)
 set(CC_USE_SE_NAPI OFF)
+set(CC_USE_SE_JSVM ON)
+
+SET(CMAKE_BUILD_TYPE "Debug")
```
Comment (qiuguohua) [inline templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt:14]: Why do we need this setup?

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 3]
```diff
@@ -51,6 +53,29 @@ void onSurfaceCreatedCB(OH_NativeXComponent* component, void* window) {
     sendMsgToWorker(cocos2d::MessageType::WM_XCOMPONENT_SURFACE_CREATED, component, window);
 }
 
+void onSurfaceHideCB(OH_NativeXComponent* component, void* window) {
+    int32_t ret;
+    char idStr[OH_XCOMPONENT_ID_LEN_MAX + 1] = {};
+    uint64_t idSize = OH_XCOMPONENT_ID_LEN_MAX + 1;
+    ret = OH_NativeXComponent_GetXComponentId(component, idStr, &idSize);
+    if(ret != OH_NATIVEXCOMPONENT_RESULT_SUCCESS) {
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp:61]: The point of getting idStr is to determine if it is valid?

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
Oh, sorry, i merged it by mistake.
@wanghui187 please answer the comment.

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt (hunk 1 above)
Comment (wanghui187) [reply templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt:71]: If jsvm is not required, there is no need to link libjsvm.so.
We need to optimize here

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt (hunk 2 above)
Comment (wanghui187) [reply templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt:14]: SET(CMAKE_BUILD_TYPE "Debug")

Comment (wanghui187) [reply templates/js-template-link/frameworks/runtime-src/proj.openharmony/CMakeLists.txt:14]: This is redundant and needs to be deleted

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 3 above)
Comment (wanghui187) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp:61]: yes，confirm whether XCompoonent is valid

**Pull request merged**